shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Rewards.hs

```diff
@@ -434,13 +515,22 @@ rewardOnePool
           pool
           (StakeShare $ fromIntegral ostake % tot)
           (StakeShare sigma)
-      f =
-        if HardForks.aggregatedRewards pp
-          then Map.insertWith (<>)
-          else Map.insert
+      poolRA = getRwdCred $ _poolRAcnt pool
       potentialRewards =
-        f (getRwdCred $ _poolRAcnt pool) lReward mRewards
-      rewards' = Map.filter (/= Coin 0) $ eval (addrsRew ◁ potentialRewards)
+        if poolRA `Set.member` addrsRew
+          then
+            Map.insertWith
+              Set.union
+              poolRA
+              (Set.singleton (Reward LeaderReward (_poolId pool) lReward))
+              mRewards
+          else mRewards
+      removeDegenerate rwds =
+        let withoutZeros = Set.filter (\rwd -> rewardAmount rwd /= Coin 0) rwds
+         in if Set.null withoutZeros
+              then Nothing
+              else Just withoutZeros
+      rewards' = Map.mapMaybe removeDegenerate potentialRewards
```

Comment:
@nc6 I know we talked about `alter`, but I've convinced myself that `mapMaybe` works better. yea?